### PR TITLE
Adding the Object Picker

### PR DESCRIPTION
**Pull request by rahul27**
GearVRf-DCO-1.0-Signed-off-by: Rahul Rudradevan
r.rudradevan@samsung.com

### CODE REVIEW

**Comment by rahul27** [comment]
This PL should replace https://github.com/Samsung/GearVRf/pull/829.

It also requires PL https://github.com/Samsung/GearVRf/pull/859.

Can be tested using https://github.com/rahul27/GearVRf-Demos/commit/f174ce44d494c6d90b7de22fdd5d6841d0a869ab

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRObjectPicker.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            GVRSceneObject.BoundingVolume bv1 = owner.getBoundingVolume();
+
+            for (int i = 0; i < picked.length; ++i)
+            {
+                GVRPickedObject hit = picked[i];
+
+                if (hit != null)
+                {
+                    GVRSceneObject sceneObj = hit.hitObject;
+                    GVRSceneObject.BoundingVolume bv2 = sceneObj.getBoundingVolume();
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRObjectPicker.java]: Might be beneficial if getBoundingVolume was updated not to always create a new Java object.

**Comment by liaxim** [comment]
@NolaDonato ?

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRObjectPicker.java (hunk 1 above)
Comment (rahul27) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRObjectPicker.java]: I can push the intersect check to native.

**Comment by rahul27** [comment]
@liaxim @NolaDonato I've made the changes.

**Comment by NolaDonato** [comment]
My overall opinion here is that BoundingVolume should be a native class whose operations are all in C++ which we access from Java. The C++ SceneObject should contain one and keep it up to date. When the user requests the bounding volume, they will always get the same pointer back in C++ and Java. We can do this in a separate pull request.

**Comment by liaxim** [comment]
Looks good to me. If @NolaDonato approves then we can merge.

**Comment by rahul27** [comment]
I found an issue with IPickEvents. Lets hold off on this merging this.

**Comment by liaxim** [comment]
Looks good to me. If @NolaDonato approves then we can merge.

**Comment by NolaDonato** [comment]
Yes, merge it.

**Pull request merged**